daemon/k8s_watcher.go

```diff
@@ -93,50 +93,59 @@ var (
 
 	k8sCM = controller.NewManager()
 
-	ruleRevCache ruleRevisionCache
+	importMetadataCache ruleImportMetadataCache
 )
 
-// ruleRevisionCache maps the unique identifier of a CiliumNetworkPolicy
-// (namespace and name) to the policy revision number of the agent's policy
-// repository at the time said rule was imported.
-type ruleRevisionCache struct {
-	mutex               lock.RWMutex
-	ruleRevisionMapping map[string]uint64
+// ruleImportMetadataCache maps the unique identifier of a CiliumNetworkPolicy
+// (namespace and name) to metadata about the importing of the rule into the
+// agent's policy repository at the time said rule was imported (revision
+// number, and if any error occurred while importing).
+type ruleImportMetadataCache struct {
+	mutex                 lock.RWMutex
+	ruleImportMetadataMap map[string]policyImportMetadata
 }
 
-func initRuleRevCache() {
-	ruleRevCache.ruleRevisionMapping = make(map[string]uint64)
+type policyImportMetadata struct {
+	revision          uint64
+	policyImportError error
 }
 
-func (r *ruleRevisionCache) upsert(cnp *cilium_v2.CiliumNetworkPolicy, revision uint64) {
+func initRuleMetadataCache() {
+	importMetadataCache.ruleImportMetadataMap = make(map[string]policyImportMetadata)
+}
+
+func (r *ruleImportMetadataCache) upsert(cnp *cilium_v2.CiliumNetworkPolicy, revision uint64, importErr error) {
 	r.mutex.Lock()
```

Comment:
create the `policyImportMetadata` object before the Lock and put the lock/unlock only on the write of the map